cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java

```diff
@@ -81,6 +83,9 @@
   private NettyHttpService httpService;
   private volatile Thread runningThread;
   private final Map<String, WorkflowActionNode> status = new ConcurrentHashMap<String, WorkflowActionNode>();
+  private volatile boolean isSuspended;
+  private ReentrantLock lock;
```

Comment:
We need the extended capabilities such as wait and notify on the condition for suspend and resume functionality. Since implicit lock does not provide those features, we need `ReentrantLock`